### ISSUE TITLE
"undefined reference to `cdotusub_'"

### ISSUE THREAD

**Issue by hughperkins** [clMathLibraries/clBLAS#184]
When building using the default options.  Result is:

```
make[2]: *** [staging/test-correctness] Error 1
make[1]: *** [tests/CMakeFiles/test-correctness.dir/all] Error 2
CMakeFiles/test-short.dir/correctness/blas-lapack.c.o: In function `cdotu':
/home/user/git/clBLAS/src/tests/correctness/blas-lapack.c:658: undefined reference to `cdotusub_'
CMakeFiles/test-short.dir/correctness/blas-lapack.c.o: In function `zdotu':
/home/user/git/clBLAS/src/tests/correctness/blas-lapack.c:673: undefined reference to `zdotusub_'
CMakeFiles/test-short.dir/correctness/blas-lapack.c.o: In function `cdotc':
/home/user/git/clBLAS/src/tests/correctness/blas-lapack.c:688: undefined reference to `cdotcsub_'
CMakeFiles/test-short.dir/correctness/blas-lapack.c.o: In function `zdotc':
/home/user/git/clBLAS/src/tests/correctness/blas-lapack.c:703: undefined reference to `zdotcsub_'
collect2: error: ld returned 1 exit status
```

Context:
- Ubuntu 14.04 64-bit
- tested two ways (wiping `build` directory, and rebuliding from scratch each time):
  - no GPU
  - NVIDIA 940M
- => both give the same error

Edit: seems I have the following blas packages installed:

```
$ dpkg -l '*blas*' | grep ii
ii  libblas-dev                                           1.2.20110419-7                                      amd64        Basic Linear Algebra Subroutines 3, static library
ii  libblas3                                              1.2.20110419-7                                      amd64        Basic Linear Algebra Reference implementations, shared library
ii  libopenblas-base                                      0.2.8-6ubuntu1                                      amd64        Optimized BLAS (linear algebra) library based on GotoBLAS2
ii  libopenblas-dev                                       0.2.8-6ubuntu1                                      amd64        Optimized BLAS (linear algebra) library based on GotoBLAS2
```

Edit2: lapack packages installed:

```
$ dpkg -l '*lapack*' | grep ii
ii  liblapack-dev                                         3.5.0-2ubuntu1                                      amd64        Library of linear algebra routines 3 - static version
ii  liblapack3                                            3.5.0-2ubuntu1                                      amd64        Library of linear algebra routines 3 - shared version
```

**Comment by TimmyLiu** [comment]
Hi I am wondering if you are seeing the same issue as reported in issue #142 that you are using openblas (/usr/lib) instead of netlib blas (/usr/lib/libblas/)?

**Comment by ghisvail** [comment]
I observe the same problem as @hughperkins when using OpenBLAS. The test suite compiles fine with standard LAPACK / BLAS.

**Comment by hughperkins** [comment]
@TimmyLiu  Yes, does look similar.  And I do have openblas installed, as you see.  I will close this one.

**Comment by tingxingdong** [comment]
This compilation bug is fix in PR #242 . U guys can take a look at and try.
